Rancher-keycloak-integration/automation_script.py

```
@@ -0,0 +1,533 @@
+            "displayNameField": saml_config["display_name_field"],
+            "groupsField": saml_config["groups_field"],
+            "idpMetadataContent": metadata_xml,
+            "rancherApiHost": self.host,  # Keep the full URL with https://
+            "spCert": certificate,
+            "spKey": private_key,
+            "uidField": saml_config["uid_field"],
+            "userNameField": saml_config["username_field"]
+        }
+        
+        # Try to get existing config first
+        response = requests.get(url, headers=self.get_headers())
```

Ruff (0.14.8)

Probable use of `requests` call without timeout

(S113)